zng/type.go

```diff
@@ -216,6 +216,15 @@ func ContainedType(typ Type) (Type, []Column) {
 	}
 }
 
+func IsUnionType(typ Type) bool {
+	switch typ.(type) {
+	case *TypeUnion:
+		return true
+	default:
+		return false
+	}
+}
+
 func IsContainerType(typ Type) bool {
 	switch typ.(type) {
 	case *TypeSet, *TypeArray, *TypeRecord:
```

Comment:
It's confusing that a union is encoded as a container but `IsContainerType((*TypeUnion)(nil))` is false. Can we make it true?

Reply:
@nwt good question. 

Since you refer to "union is encoded as a container" I think the premise of your comment is that `zng.IsContainerType()` is a question about the bzng encoding of a type. If that's the case, then I agree that answering "false" to `zng.IsContainerType(union)` is confusing - in fact it's not just confusing, it's incorrect. 

However (and I think more importantly), if `zng.IsContainerType()` is a question about the bzng encoding of the type, then it should presumably be moved elsewhere (bzngio?) as it's not the zng package's responsibility to answer questions about how bzng encoding works.

A counter-argument could be made that `IsContainerType` is a question about the type itself rather than of its bzng encoding (as per the type-level distinction "primitive" vs "container" that we've talked about elsewhere). This interpretation is supported by the use of `IsContainerType()` (via `Value.IsContainer()`) in some zql procs. Under this interpretation answering false to `zng.IsContainerType(union)` is correct. 

I think a clean resolution could be to have both a `zng.IsContainerType()` (that answers the question about the type itself) and a `bzngio.IsContainerEncoded()` (or some better name) that answers the question about the physical layer encoding, and use the right one in the right place. For clarity we might also want to consider using a different terminology for on-the-wire bzng "containers" vs type-level "containers", but that's a broader question.

Long story short: @nwt let me know how I should proceed and I will do it in this PR or a separate one as appropriate.

Reply:
Another element for context here: I was just reminded of `zng.ContainedType()`, that seems to line up with the second interpretation above...

Reply:
My view is that package zng directly reflects the BZNG spec and that ought to continue unless there's a good reason. Diverging here adds a surprising quirk, and I'm not seeing the good reason.

> A counter-argument could be made that IsContainerType is a question about the type itself rather than of its bzng encoding (as per the type-level distinction "primitive" vs "container" that we've talked about elsewhere). Under this interpretation answering false to zng.IsContainerType(union) is correct.

The flaw I see in this argument is that a union value does in fact contain another value. That's my definition of a container. 😃

Reply:
> The flaw I see in this argument is that a union value does in fact contain another value. 

I disagree, but this may be a bikeshed. 

Regardless, how would you like me to proceed? I can make the change you request, but if so please let me know what (if anything) should be done about `zng.ContainedType`. If a union type is a container, then containedType should return the "inner" type of the union type, which can't be done based on the type alone...